src/interfaces/permissions/request/method.ts

```diff
@@ -0,0 +1,40 @@
+import type { InterfaceMethod } from '../../types';
+import type { JsonPermissionsRequest } from './types';
+
+import { MessageResult } from '../../../response';
+import { PermissionsRequest } from './message';
+
+export const processPermissionsRequest: InterfaceMethod = async (
+  ctx,
+  message,
+  messageStore,
+  didResolver
+): Promise<MessageResult> => {
+  const request = new PermissionsRequest(message as JsonPermissionsRequest);
+
+  if (ctx.tenant !== request.grantedBy && ctx.tenant !== request.grantedTo) {
```

Comment:
Sanity: why does `ctx.tenant` have to be thge same as `request.grantedTo`?

Also in delegation scenario, how does Carol make request to Bob to get access to Alice's DWN?

According to [spec](https://identity.foundation/decentralized-web-node/spec/#request)
Bob = grantedBy
Carol = grantedTo
Alice = ???

Am I looking at an outdated spec?

Reply:
> Sanity: why does `ctx.tenant` have to be the same as `request.grantedTo`?

the tenant should be either the grantee or grantor, otherwise it seems a bit strange for the person to be storing the permission.

> Also in delegation scenario, how does Carol make request to Bob to get access to Alice's DWN?

Carol would actually make a request to Alice's DWN directly. I don't think we had ever considered requesting delegated permissions. Interesting thought. We should discuss later on and incorporate if needed